modules/@apostrophecms/i18n/index.js

```diff
@@ -407,6 +459,30 @@ module.exports = {
         req.baseUrlWithPrefix = `${self.apos.page.getBaseUrl(req)}${self.apos.prefix}`;
         req.absoluteUrl = req.baseUrlWithPrefix + req.url;
         req.prefix = `${req.baseUrlWithPrefix}${self.locales[req.locale].prefix || ''}`;
+      },
+      // Returns an Express route suitable for use in a module
+      // like a piece type or the page module. The returned route will
+      // expect req.params._id and req.params.toLocale and redirect,
+      // if possible, to the corresponding version in toLocale.
+      toLocaleRouteFactory(module) {
```

Comment:
Trying out the factory pattern as a response to the problem of many routes being duplicated between piece-type and the page module, which don't and shouldn't share a lineage but do have some cases of routes that aren't going to differ between them.